crates/router/src/connector/bankofamerica/transformers.rs

```diff
@@ -2668,25 +2679,56 @@ impl TryFrom<types::RefundsResponseRouterData<api::RSync, BankOfAmericaRsyncResp
     fn try_from(
         item: types::RefundsResponseRouterData<api::RSync, BankOfAmericaRsyncResponse>,
     ) -> Result<Self, Self::Error> {
-        match item.response {
-            BankOfAmericaRsyncResponse::RsyncApplicationResponse(rsync_response) => Ok(Self {
-                response: Ok(types::RefundsResponseData {
-                    connector_refund_id: rsync_response.id,
-                    refund_status: enums::RefundStatus::from(
-                        rsync_response.application_information.status,
-                    ),
-                }),
-                ..item.data
-            }),
-            BankOfAmericaRsyncResponse::ErrorInformation(error_response) => Ok(Self {
-                status: item.data.status,
-                response: Ok(types::RefundsResponseData {
-                    refund_status: common_enums::RefundStatus::Pending,
-                    connector_refund_id: error_response.id.clone(),
-                }),
-                ..item.data
+        let response = match item
+            .response
+            .application_information
+            .and_then(|application_information| application_information.status)
+        {
+            Some(status) => {
+                let refund_status: common_enums::RefundStatus =
+                    enums::RefundStatus::from(status.clone());
+                if utils::is_refund_failure(refund_status) {
+                    if status == BankofamericaRefundStatus::Voided {
+                        Err(types::ErrorResponse::from((
+                            &Some(BankOfAmericaErrorInformation {
+                                message: Some(consts::REFUND_VOIDED.to_string()),
+                                reason: None,
+                            }),
+                            &None,
+                            None,
+                            item.http_code,
+                            item.response.id.clone(),
+                        )))
```

Comment:
Why do we need to do this? please have relevant comments explaining why this is done.